include/swift/AST/Attr.h

```diff
@@ -1508,6 +1508,19 @@ class DifferentiableAttr final
   static bool classof(const DeclAttribute *DA) {
     return DA->getKind() == DAK_Differentiable;
   }
+
+  bool parametersMatch(const DifferentiableAttr &other) const {
+    auto a = getParsedParameters();
+    auto b = other.getParsedParameters();
+    if (a.size() != b.size())
+      return false;
+
+    for (size_t i = 0; i < b.size(); ++i) {
```

Comment:
The local convention is to use `unsigned` and use a temporary for count.
```suggestion
    for (unsigned i = 0, n = b.size(); i != n; ++i) {
```